desktop/libs/azure/src/azure/abfs/abfs.py

```diff
@@ -572,6 +573,16 @@ def rename(self, old, new):
     Renames a file
     """
     headers = {'x-ms-rename-source': '/' + urllib_quote(Init_ABFS.strip_scheme(old))}
+
+    # Required to sign the header with SAS token for RAZ
+    if RAZ.IS_ENABLED.get():
+      raz_client = AdlsRazClient(username=self._user)
+      source_url = self._url + normpath('/' + headers['x-ms-rename-source'].lstrip('/'))
+
+      response = raz_client.get_url(action='PUT', path=source_url)
+      if response and response.get('token'):
+        headers['x-ms-rename-source'] += '?' + response['token']
+
```

Comment:
Can this live in the raz client? (i.e. we provide an `old` that is signed?

Reply:
Decoupled SAS token fetching in the raz http client itself so that we can reuse code.